### PR TITLE
Migrate Figure.inset tests to use dvc

### PR DESCRIPTION
**Pull request by willschlitzer**
As discussed in #1131, this moves the reference figures in `fig.inset` to use `dvc`.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
> This pull request keeps `@check_figures_equal` for `test_inset_context_manager()` as it is testing the use of the context manager.

I think we still should refactor this test.

**Comment by willschlitzer** [comment]
> > This pull request keeps `@check_figures_equal` for `test_inset_context_manager()` as it is testing the use of the context manager.
> 
> 
> 
> I think we still should refactor this test.

What should I use for the reference image to test the context manager?

**Comment by seisman** [comment]
> Test that the inset context manager works and, once closed, plotting elements are added to the larger figure.

The main purpose of this test is to check if `basemap(rose="jTR+w3c")` plots the rose in the large figure, not in the inset map. So you can simply remove the reference image.

**Comment by willschlitzer** [comment]
> > Test that the inset context manager works and, once closed, plotting elements are added to the larger figure.
> 
> The main purpose of this test is to check if `basemap(rose="jTR+w3c")` plots the rose **in the large figure, not in the inset map**. So you can simply remove the reference image.

Updated it to use a reference image.

File: pygmt/tests/test_inset.py [hunk 1]
```diff
@@ -1,42 +1,32 @@
 """
 Tests for the inset function.
 """
+import pytest
 from pygmt import Figure
 from pygmt.helpers.testing import check_figures_equal
```
Comment (weiji14) [inline pygmt/tests/test_inset.py:6]:
```suggestion
```

**Review by weiji14 — APPROVED** [review]
Just need to remove the `check_figures_equal` import and it should be good to merge I think.

**Pull request merged**